functions/minsurf.go

```diff
@@ -0,0 +1,245 @@
+			dvLdy := (vTL - vLL) / ms.hy
+			dvTdx := (vTR - vTL) / ms.hx
+			dvTdy := (vTR - vLR) / ms.hy
+
+			fL := math.Sqrt(1 + dvLdx*dvLdx + dvLdy*dvLdy)
+			fT := math.Sqrt(1 + dvTdx*dvTdx + dvTdy*dvTdy)
+			area += fL + fT
+
+			if grad != nil {
+				if i > 0 {
+					if j > 0 {
+						grad[ms.index(i, j)] -= (dvLdx/ms.hx + dvLdy/ms.hy) / fL
+					}
+					if j < ny-2 {
+						grad[ms.index(i, j+1)] += (dvLdy/ms.hy)/fL - (dvTdx/ms.hx)/fT
+					}
+				}
+				if i < nx-2 {
+					if j > 0 {
+						grad[ms.index(i+1, j)] += (dvLdx/ms.hx)/fL - (dvTdy/ms.hy)/fT
+					}
+					if j < ny-2 {
+						grad[ms.index(i+1, j+1)] += (dvTdx/ms.hx + dvTdy/ms.hy) / fT
+					}
+				}
+			}
+		}
+
+	}
+	cellSize := 0.5 * ms.hx * ms.hy
+	area *= cellSize
+	for i := range grad {
+		grad[i] *= cellSize
+	}
+
+	return area
+}
+
+// InitX returns a starting location for the minimization problem. Length of
+// the returned slice is (nx-2)×(ny-2).
+func (ms *MinimalSurface) InitX() []float64 {
+	nx, ny := ms.GridSize()
+	x := make([]float64, (nx-2)*(ny-2))
+	for i := 1; i < nx-1; i++ {
+		for j := 1; j < ny-1; j++ {
+			x[ms.index(i, j)] = (ms.left[j] + ms.bottom[i]) / 2
+		}
+	}
+	return x
+}
+
+// ExactX returns the exact solution to the minimization problem. Length of the
+// returned slice is (nx-2)*(ny-2).
+func (ms *MinimalSurface) ExactX() []float64 {
+	nx, ny := ms.GridSize()
+	v := make([]float64, (nx-2)*(ny-2))
+	for i := 1; i < nx-1; i++ {
+		x := -0.5 + float64(i)*ms.hx
+		for j := 1; j < ny-1; j++ {
+			y := -0.5 + float64(j)*ms.hy
+			v[ms.index(i, j)] = ms.ExactSolution(x, y)
+		}
+	}
+	return v
+}
+
+// ExactSolution returns the value of the exact solution to the minimal surface
+// problem at (x,y). The exact solution is defined as
+//  F_exact(x,y) = U^2(x,y) - V^2(x,y),
+// where U and V are the unique solutions to the equations
+//  x =  u + uv^2 - u^3/3,
+//  y = -v - u^2v + v^3/3.
+func (ms *MinimalSurface) ExactSolution(x, y float64) float64 {
+	var u = [2]float64{x, -y}
+	var f [2]float64
+	var jac [2][2]float64
+	for k := 0; k < 100; k++ {
+		f[0] = u[0] + u[0]*u[1]*u[1] - u[0]*u[0]*u[0]/3 - x
+		f[1] = -u[1] - u[0]*u[0]*u[1] + u[1]*u[1]*u[1]/3 - y
+		fNorm := math.Hypot(f[0], f[1])
+		if fNorm < 1e-13 {
+			break
+		}
+		jac[0][0] = 1 + u[1]*u[1] - u[0]*u[0]
+		jac[0][1] = 2 * u[0] * u[1]
+		jac[1][0] = -2 * u[0] * u[1]
+		jac[1][1] = -1 - u[0]*u[0] + u[1]*u[1]
+		det := jac[0][0]*jac[1][1] - jac[0][1]*jac[1][0]
+		u[0] -= (jac[1][1]*f[0] - jac[0][1]*f[1]) / det
+		u[1] -= (jac[0][0]*f[1] - jac[1][0]*f[0]) / det
+	}
+	return u[0]*u[0] - u[1]*u[1]
+}
+
+// GridSize returns the size of the underlying rectilinear grid.
+func (ms *MinimalSurface) GridSize() (nx, ny int) {
+	return len(ms.bottom), len(ms.left)
+}
+
+// Steps returns the spatial step sizes of the underlying rectilinear grid.
+func (ms *MinimalSurface) Steps() (hx, hy float64) {
+	return ms.hx, ms.hy
+}
+
+// WriteGnuplot saves in Gnuplot format the surface represented by the vector
```

Comment:
* MeshGrid seems to be rather a function defined on a grid, wouldn't GridFunction be a better name?
* It would be more natural for Dims and At to use x, y rather than y, x.

I will remove WriteGnuplot().